origins_generators/sources/delimited.py

```diff
@@ -75,47 +84,35 @@ def setup(self):
         if not self.options.columns:
             self.options.columns = _header
 
-    def parse_file(self):
-        uri = self.options.uri
-
-        name = os.path.splitext(os.path.basename(uri))[0]
-
-        if not utils.is_remote(self.options.uri):
-            uri = os.path.abspath(uri)
-
-        return {
-            'origins:id': name,
-            'prov:label': utils.prettify_name(name),
-            'prov:type': 'File',
-            'uri': uri,
-        }
-
     def parse_columns(self):
         columns = []
-
         for i, name in enumerate(self.options.columns):
-            column = {
-                'origins:id': name,
-                'prov:label': name,
-                'prov:type': 'Column',
-                'index': i,
-            }
-
-            columns.append(column)
-
+            columns.append(name)
         return columns
 
     def parse(self):
-        file = self.parse_file()
-        self.document.add('entity', file)
-
-        columns = self.parse_columns()
-
-        for column in columns:
-            self.document.add('entity', column)
-
-            self.document.add('wasInfluencedBy', {
-                'prov:influencer': file,
-                'prov:influencee': column,
-                'prov:type': 'origins:Edge',
-            })
+        FIELDS = self.parse_columns()
+        with open(self.options.uri, 'rU', newline='') as f:
+            yield [
+                'operation',
+                'domain',
+                'entity',
+                'attribute',
+                'value',
+                'valid_time'
+            ]
+
+            reader = csv.DictReader(f, fieldnames=FIELDS,
+                                    delimiter=self.options.delimiter)
+            next(reader)
+            for line in reader:
+                for key, value in line.items():
+                    if key != FIELDS[0]:
+                        yield [
```

Comment:
Python lists are mutable, while tuples are *immutable*. Since these facts are not expected to change downstream they should be yielded as tuples, not lists. This applies to all the generators.